src/util/system.cpp

```diff
@@ -863,6 +863,32 @@ std::vector<util::SettingsValue> ArgsManager::GetSettingsList(const std::string&
     return util::GetSettingsList(m_settings, m_network, SettingName(arg), !UseDefaultSection(arg));
 }
 
+void ArgsManager::logArgsPrefix(
+    const std::string& prefix,
+    const std::string& section,
+    const std::map<std::string, std::vector<util::SettingsValue>>& args) const
+{
+    std::string section_str = section.empty() ? "" : "[" + section + "] ";
+    for (const auto& arg : args) {
+        for (const auto& value : arg.second) {
+            Optional<unsigned int> flags = GetArgFlags('-' + arg.first);
+            if (flags) {
+                std::string value_str = (*flags & DONT_LOG) ? "****" : value.write();
+                LogPrintf("%s %s%s=%s\n", prefix, section_str, arg.first, value_str);
+            }
+        }
+    }
+}
+
+void ArgsManager::LogArgs() const
+{
+    LOCK(cs_args);
+    for (const auto& section : m_settings.ro_config) {
+        logArgsPrefix("config-file arg:", section.first, section.second);
+    }
+    logArgsPrefix("command-line arg:", "", m_settings.command_line_options);
```

Comment:
s/command-line/Command-line/ (first letter of logs are usually capitalized)

Reply:
Thanks, done, separate commit 717099eef1cc5670180a58d0fec6d027a1e2f95d. I'll squash after review (or, at least before merge). Current tree hash:
```
$ curl -s https://api.github.com/repos/bitcoin/bitcoin/commits/717099eef1cc5670180a58d0fec6d027a1e2f95d | jq .commit.tree.sha
"d38bc36d10f9ebbd6674a95c95eb9e17c7c05fdc"
$ 
```
(We can do this again with the new commit hash after squashing to ensure I haven't accidentally changed the reviewed code.)

Reply:
> I'll squash after review (or, at least before merge)

It is better to squash before review as a merge commit should contain ACKs with the latest commit hash.